proto/pulumi/language.proto

```diff
@@ -23,290 +23,561 @@ package pulumirpc;
 
 option go_package = "github.com/pulumi/pulumi/sdk/v3/proto/go;pulumirpc";
 
-// LanguageRuntime is the interface that the planning monitor uses to drive execution of an interpreter responsible
-// for confguring and creating resource objects.
+// The LanguageRuntime service defines a standard interface for [language hosts/runtimes](languages). At a high level, a
+// language runtime provides the ability to execute programs, install and query dependencies, and generate code for a
+// specific language.
 service LanguageRuntime {
-    // `Handshake` is the first call made by the engine to a language host. It is used to pass the 
-    // engine's address to the language host so that it may establish its own connections back,
-    // and to establish protocol configuration that will be used to communicate between the two parties. 
+    // `Handshake` is the first call made by the engine to a language host. It is used to pass the engine's address to
+    // the language host so that it may establish its own connections back, and to establish protocol configuration that
+    // will be used to communicate between the two parties.
     rpc Handshake(LanguageHandshakeRequest) returns (LanguageHandshakeResponse) {}
 
-    // GetRequiredPlugins computes the complete set of anticipated plugins required by a program.
+    // `GetRequiredPlugins` computes the complete set of anticipated [plugins](plugins) required by a Pulumi program. It
+    // is intended to be used to pre-install plugins before running a program with [](pulumirpc.LanguageRuntime.Run), to
+    // avoid the need to install them on-demand in response to [resource registrations](resource-registration) sent back
+    // from the running program to the engine.
+    //
+    // :::{important}
+    // The use of `GetRequiredPlugins` is deprecated in favour of [](pulumirpc.LanguageRuntime.GetRequiredPackages),
+    // which returns more granular information about which plugins are required by which packages.
+    // :::
     rpc GetRequiredPlugins(GetRequiredPluginsRequest) returns (GetRequiredPluginsResponse) {
         option deprecated = true;
     }
-    // GetRequiredPackages computes the complete set of anticipated packages required by a program.
+
+    // `GetRequiredPackages` computes the complete set of anticipated [packages](pulumirpc.PackageDependency) required
+    // by a program. It is intended to be used to pre-install packages before running a program with
```

Comment:
It's also used for us to build expected providers for import. So that if we see `pulumi import my-param-prov:index:typ` we can know what `my-param-prov` is.